layer/executables/layer_runtime.py

```diff
@@ -143,6 +194,32 @@ def _get_function_output(metadata: Dict[str, Any]) -> _FunctionOutput:
     return _FunctionOutput(type, name)
 
 
+def _get_function_source_digest(metadata: Dict[str, Any]) -> str:
+    source_digest = metadata.get("function", {}).get("source", {}).get("digest")
+    if source_digest is None:
+        raise LayerFunctionRuntimeError("function metadata missing source digest")
+    return source_digest
+
+
+def _upload_model_artifacts(model_dir: Path, config: TrainStorageConfiguration) -> None:
+    import boto3
+
+    s3_config = {
+        "aws_access_key_id": config.credentials.access_key_id,
```

Comment:
Where are we going to get these from? Is there a `.layer/config` in the runner? Maybe populated by the `login()` function?

Reply:
> Where are we going to get these from?

The credentials are generated in the backend. Access by the API key.